### ISSUE TITLE
NPM version next

### ISSUE THREAD

**Issue by reiz** [versioneye/versioneye-core#117]
It seems that `next` is special keyword or version for NuGet packages. Can you please double check that how to handle that correctly? 

![screen shot 2017-07-20 at 09 36 28](https://user-images.githubusercontent.com/652130/28406141-91fad1d0-6d2f-11e7-8592-cb7ba21b4b05.png)

**Comment by reiz** [comment]
This is NOT related to NuGet. It's related to NPM.

**Comment by timgluz** [comment]
I did some research, it's called `dist-tag` - https://docs.npmjs.com/cli/dist-tag and this tag can be any string, can have any meaning and controlled by developer;

We could update crawler to save `dist-tags` for each version and remove old ones ; so we can check up versions by matching tags;

**Comment by timgluz** [comment]
Here's Yarn issue which discusses how Yarn and NPM solved it differently: https://github.com/yarnpkg/yarn/issues/1690

**Comment by reiz** [comment]
@timgluz are the `dist-tags` available through the NPMRegistry? Then we should crawl them. They include all we need:

```
{ latest: '2.0.10',
  next: '2.2.0-dev.20161129',
  beta: '2.0.0',
  rc: '2.1.1',
  insiders: '2.0.6-insiders.20161017' }
```

With that we could easily replace `next` and `latest` through a valid version number.

**Comment by timgluz** [comment]
Yes, API exposes `dist-tags`. example: https://registry.npmjs.com/uglifyjs

I will update crawler to save each tag for each version and then i will update parser to add support comparing tags;

**Comment by timgluz** [comment]
Ok, ready for reviews;

**Comment by timgluz** [comment]
Done!

**Comment by reiz** [comment]
@timgluz Why it's open again? Something wrong?

**Comment by timgluz** [comment]
no, it was unclosed ticket and i accidentally clicked on "close and comment"
